RecoVertex/BeamSpotProducer/plugins/BeamSpotAnalyzer.cc

```diff
@@ -110,61 +137,62 @@ void BeamSpotAnalyzer::endLuminosityBlock(const edm::LuminosityBlock& lumiSeg, c
+    edm::LogPrint("BeamSpotAnalyzer") << bs << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << "[BeamFitter] fit done. \n" << std::endl;
   } else {  // Fill in empty beam spot if beamfit fails
     reco::BeamSpot bs;
     bs.setType(reco::BeamSpot::Fake);
-    std::cout << "\n Empty Beam spot fit" << std::endl;
-    std::cout << " for runs: " << ftmprun0 << " - " << ftmprun << std::endl;
-    std::cout << " for lumi blocks : " << LSRange.first << " - " << LSRange.second << std::endl;
-    std::cout << " lumi counter # " << countLumi_ << std::endl;
-    std::cout << bs << std::endl;
-    std::cout << "[BeamFitter] fit failed \n" << std::endl;
-    //accumulate more events
-    // dissable this for the moment
+    edm::LogPrint("BeamSpotAnalyzer") << "\n Empty Beam spot fit" << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << " for runs: " << ftmprun0 << " - " << ftmprun << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << " for lumi blocks : " << LSRange.first << " - " << LSRange.second << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << " lumi counter # " << countLumi_ << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << bs << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << "[BeamFitter] fit failed \n" << std::endl;
+    // accumulate more events
+    // disable this for the moment
     //resetFitNLumi_ += 1;
-    //std::cout << "reset fitNLumi " << resetFitNLumi_ << std::endl;
+    //edm::LogPrint("BeamSpotAnalyzer") << "reset fitNLumi " << resetFitNLumi_ << std::endl;
   }
 
   if (resetFitNLumi_ > 0 && countLumi_ % resetFitNLumi_ == 0) {
     std::vector<BSTrkParameters> theBSvector = theBeamFitter->getBSvector();
-    std::cout << "Total number of tracks accumulated = " << theBSvector.size() << std::endl;
-    std::cout << "Reset track collection for beam fit" << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << "Total number of tracks accumulated = " << theBSvector.size() << std::endl;
+    edm::LogPrint("BeamSpotAnalyzer") << "Reset track collection for beam fit" << std::endl;
     theBeamFitter->resetTrkVector();
     theBeamFitter->resetLSRange();
     theBeamFitter->resetCutFlow();
     theBeamFitter->resetRefTime();
     theBeamFitter->resetPVFitter();
     countLumi_ = 0;
     // reset counter to orginal
-    resetFitNLumi_ = Org_resetFitNLumi_;
+    resetFitNLumi_ = org_resetFitNLumi_;
   }
 }
 
 void BeamSpotAnalyzer::endJob() {
-  std::cout << "\n-------------------------------------\n" << std::endl;
-  std::cout << "\n Total number of events processed: " << ftotalevents << std::endl;
-  std::cout << "\n-------------------------------------\n\n" << std::endl;
+  edm::LogPrint("BeamSpotAnalyzer") << "\n-------------------------------------\n" << std::endl;
+  edm::LogPrint("BeamSpotAnalyzer") << "\n Total number of events processed: " << ftotalevents << std::endl;
+  edm::LogPrint("BeamSpotAnalyzer") << "\n-------------------------------------\n\n" << std::endl;
```

Comment:
Is the std::endl needed when edm::LogPrint is used? I thought the LogPrint takes care of the new line.